File: app/src/main/java/com/pavel/webviewcallback/web/WebViewJsInterface.java
package com.pavel.webviewcallback.web;

import android.content.Context;
import android.webkit.JavascriptInterface;
import android.webkit.WebView;
import android.widget.Toast;

/**
 * Created by anshul on 18/04/17.
 */

public class WebViewJsInterface {

  private WebView webView;
  private Context context;

  public WebViewJsInterface(Context context, WebView webView) {
    this.webView = webView;
    this.context=context;
  }

  @JavascriptInterface
  public void tokenStatus(boolean value, String message) {
    Toast.makeText(context, ""+value, Toast.LENGTH_LONG).show();
    Toast.makeText(context, ""+message, Toast.LENGTH_LONG).show();
  }

  @JavascriptInterface
  public void contentChange(String path, boolean isPremium) {
    Toast.makeText(context, ""+isPremium, Toast.LENGTH_LONG).show();
    Toast.makeText(context, ""+path, Toast.LENGTH_LONG).show();
  }

  @JavascriptInterface
  public void eventExecute(String params) {
    Toast.makeText(context, "event "+params, Toast.LENGTH_LONG).show();
  }

}

File: app/src/main/java/com/pavel/webviewcallback/web/WebViewJsInterface.kt
package com.pavel.webviewcallback.web

import android.content.Context
import android.webkit.JavascriptInterface
import android.widget.Toast


class WebViewJsInterface(private val context: Context) {
    @JavascriptInterface
    fun tokenStatus(value: Boolean, message: String) {
        Toast.makeText(context, "" + value, Toast.LENGTH_LONG).show()
        Toast.makeText(context, "" + message, Toast.LENGTH_LONG).show()
    }

    @JavascriptInterface
    fun contentChange(path: String, isPremium: Boolean) {
        Toast.makeText(context, "" + isPremium, Toast.LENGTH_LONG).show()
        Toast.makeText(context, "" + path, Toast.LENGTH_LONG).show()
    }

    @JavascriptInterface
    fun eventExecute(params: String, value:String) {
        Toast.makeText(context, "event $params", Toast.LENGTH_LONG).show()
        Toast.makeText(context, "value $value", Toast.LENGTH_LONG).show()
    }
}


// class WebViewJsInterface(private val context: Context) {
//    @JavascriptInterface
//    fun eventExecute(params: String?) {
//
//        //Logger.d("PartnerServiceDebugLog params: $params")
//
//        /*val bundle = Bundle()
//
//        if (!TextUtils.isEmpty(params)) {
//
//        val paramArray = params?.split(",")?.toTypedArray()
//
//        bundle.putString(paramArray?.getOrNull(0), paramArray?.getOrNull(1))
//
//        Application.logSDKEvent(partnerSlug, name, bundle)
//
//        }*/
//
//    }
//    @JavascriptInterface
//    fun tokenStatus(value: String?, message: String?) {
//        show("token is ${value}")
//        show("Message is ${message}")
//    }
//
//    @JavascriptInterface
//    fun contentChange(path: String?, isPremium: String?) {
//        show("Path: ${path}")
//        show("Premium: ${isPremium}")
//    }
//
//    fun show(message: String){
//        Toast.makeText(context , message , Toast.LENGTH_SHORT ).show()
//    }
//
//}
//
